src/nnet/nnet-linear-transform.h

```diff
@@ -80,10 +83,17 @@ class LinearTransform : public UpdatableComponent {
 
     //
     // Initialize trainable parameters,
-    //
-    // Gaussian with given std_dev (mean = 0),
-    linearity_.Resize(OutputDim(), InputDim());
-    RandGauss(0.0, param_stddev, &linearity_);
+    //  if Xavier_flag=1, use the “Xavier” initialization
+    if(xavier_flag){
```

Comment:
Why do you need this 'special' initialization hard-coded? 
1) Xavier Glorot's initialization was for the <Tanh> networks only.
2) It is possible to set the mean/variance of the initialization externally.
Was it helpful in some way?